torch/nn/modules/module.py

```diff
@@ -498,6 +541,33 @@ def register_backward_hook(self, hook):
         self._backward_hooks[handle.id] = hook
         return handle
 
+    @classmethod
+    def register_global_forward_pre_hook(cls, hook):
+        r"""Registers a forward pre-hook common to all modules.
+
+        The hook will be called every time before :func:`forward` is invoked.
+        It should have the following signature::
+
+            hook(module, input) -> None or modified input
+
+        The input contains only the positional arguments given to the module.
+        Keyword arguments won't be passed to the hooks and only to the ``forward``.
+        The hook can modify the input. User can either return a tuple or a
+        single modified value in the hook. We will wrap the value into a tuple
+        if a single value is returned(unless that value is already a tuple).
+
+        This hook has precedence over the specific module hooks registered with
```

Comment:
Why is this one executed before the local hooks while the two others are after?

Reply:
Actually this is one point I wanted to discuss, the ordering of execution.

Do you think that having the global before all the local ones is a reasonable behavior?
I will adjust it.